design-principles.md

```diff
@@ -38,3 +38,4 @@ The Tekton API currently contains some Kubernetes-isms. These should be treated
 1. In TEPs, discuss how the proposal affects [conformance](https://github.com/tektoncd/community/blob/main/teps/0012-api-spec.md).
 As mentioned in the conformance document, newly added API fields shouldn't be required for Tekton conformance, but can
 transition to becoming required over time.
+1. As much as possible, avoid feature flags that make Tasks and Pipelines actually behave differently. It should be possible to take a `Task/Pipeline` and reuse it across systems without requiring that a controller be configured differently to make it work. 
```

Comment:
> Not sure I understand this, can you give some examples? Is this as opposed to flags configuring the behavior of a TaskRun or PipelineRun?

This came out of a discussion between Jerop, Christie and myself while working on another TEP. I thought that this was a principle that we always followed in hindsight, just didn't document it formally. But I will let @jerop or @bobcatfish comment otherwise.

Reply:
> NIT: perhaps drop the "as much as possible" part, and add "Exceptions should be discussed with in the TEP and agreed upon with the Tekton community".

Will do! I agree. Thanks for the suggestion.